src/views/box_view.rs

```diff
@@ -212,12 +213,25 @@ impl<T: View> BoxView<T> {
 impl<T: View> ViewWrapper for BoxView<T> {
     wrap_impl!(self.view: T);
 
+    fn wrap_draw(&self, printer: &Printer) {
+        let mut printer = printer.clone();
+        // Set values hard here, since cropping would allow smaller size if fixed or at least set
+        printer.size.x = self.size.x.result((printer.size.x, printer.size.x));
+        printer.size.y = self.size.y.result((printer.size.y, printer.size.y));
+        self.view.draw(&printer);
+        self.view.draw(&printer);
+    }
+
     fn wrap_required_size(&mut self, req: Vec2) -> Vec2 {
         // This is what the child will see as request.
         let req = self.size.zip_map(req, SizeConstraint::available);
 
         // This is the size the child would like to have.
-        let child_size = self.view.required_size(req);
+        // Given the constraints of our box.
+        let child_size = self.view.required_size(Vec2::new(
+            self.size.x.result((req.x, req.x)),
+            self.size.y.result((req.y, req.y)),
```

Comment:
Here too you could so `self.size.zip_map(req, |c, r| c.result((r, r)))`.

Reply:
Changed, to this. Again thanks for the suggestion :+1: